### DISCUSSION TITLE
restic restore of externally created snapshot fails in Docker due to lchown: invalid argument

### DISCUSSION THREAD

**Discussion by virus4sepp** [garethgeorge/backrest#610] [Q&A] [answered]
Hello,

we installed backrest on Docker and also installed it on an proxmox container. The Backup is done seperatly without using backrest. I can set everything up in backrest and see the repositorys, snapshots and files. I also can brows the snapshots. But if i like to restore a file it get an erro code: " failed: exit code 1: restore failed"

what is the problem?

Thanks for your help

**Comment by garethgeorge** [comment]
You should see a "view logs" button next to the restore operation -- click this for more detailed output re: why your restore command failed.

**Reply by virus4sepp** [reply]
Hello,
thats what the log says

Thank you


`[repo-manager] 08:05:39.740Z	debug	restore snapshot	{"repo": "XXX", "snapshot": "db0f1507c9640a9c5c0298aef81001da37cf3302d8627f3e17097bd784058e9b", "target": "/root/Downloads/restic-restore-2024-12-16T08-05-39"}
[restic] 
[restic] command: /bin/restic restore --json db0f1507c9640a9c5c0298aef81001da37cf3302d8627f3e17097bd784058e9b:/hdd-zfs/subvol-100-disk-1/Dokumente/00010_Interne Buerotaetigkeiten/10_Dokumente/15 Schriftverkehr Standard -o sftp.args=-oBatchMode=yes --target /root/Downloads/restic-restore-2024-12-16T08-05-39 --include Auftragsbestaetigung Kalibrierung Messgeraete_1119.pdf
[restic] {"message_type":"error","error":{"message":"lchown /root/Downloads/restic-restore-2024-12-16T08-05-39/Auftragsbestaetigung Kalibrierung Messgeraete_1119.pdf: invalid argument"},"during":"restore","item":"/Auftragsbestaetigung Kalibrierung Messgeraete_1119.pdf"}
[restic] {"message_type":"summary","total_files":1,"files_restored":1,"total_bytes":69372,"bytes_restored":69372}
[restic] Fatal: There were 1 errors
[restic] 
[tasklog] 08:05:41.629Z	error	restore snapshot "db0f1507c9640a9c5c0298aef81001da37cf3302d8627f3e17097bd784058e9b" in repo "XXX"	task failed	{"error": "restore failed: restore snapshot \"db0f1507c9640a9c5c0298aef81001da37cf3302d8627f3e17097bd784058e9b:/hdd-zfs/subvol-100-disk-1/Dokumente/00010_Interne Buerotaetigkeiten/10_Dokumente/15 Schriftverkehr Standard\" for repo XXX: command \"/bin/restic restore --json db0f1507c9640a9c5c0298aef81001da37cf3302d8627f3e17097bd784058e9b:/hdd-zfs...\" failed: exit code 1: restore failed", "duration": 1.889196112}`

**Comment by garethgeorge** [comment, marked as answer]
Hmm -- it looks like you're running into something related to https://github.com/restic/restic/issues/655 . At the time, the reported issue is a user trying to run a restore that chown's files without root.

In docker, it's more likely that this is a problem with how user IDs are mapped into the container. 

Backrest is supported in docker because it's by far the easiest way to get started with backups -- and I'd rather people have easy to setup backups than no backups at all :) but there are a number of footguns especially with user ID mapping and filesystem permissions when using a virtual filesystem layer in between restic the dataset.

My suspicion is that there are uid's / gid's specified in the backup that don't exist in the docker container and so backrest is unable to restore those permissions. You'll probably need a bare-metal (non-dockerized install) of backrest to restore your externally created snapshots. Backrest should work fine in this sort of setup for any backups created inside the container.

**Reply by virus4sepp** [reply]
hello gareth,
thanks for your response. i can confirm that. i tryed it on proxmox lxc container, it also dosnt work with unprivileged container but it worked when the container is privileged.

thanks a lot.
greetings christoph
